pkg/apis/feeds/v1alpha1/feed_validation.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"strings"
+)
+
+func (f *Feed) Validate() *apis.FieldError {
+	return f.Spec.Validate().ViaField("spec")
+}
+
+func (fs *FeedSpec) Validate() *apis.FieldError {
+	if err := fs.Trigger.Validate(); err != nil {
+		return err.ViaField("trigger")
+	}
+	if err := fs.Action.Validate(); err != nil {
+		return err.ViaField("action")
+	}
+	return nil
```

Comment:
Do we want to aggregate these errors before returning?

If so, we might need a change to `knative/pkg` first to support a "combine" method. Feel free to defer.

Reply:
Definitely defer this, since it looks like apis.FieldError can't support reporting different types of errors in one rollup.